src/backend_model.h

```diff
@@ -122,9 +123,12 @@ class TritonModel : public Model {
   // Backend used by this model.
   std::shared_ptr<TritonBackend> backend_;
 
-  // The model instances for this model.
-  std::vector<std::unique_ptr<TritonModelInstance>> instances_;
-  std::vector<std::unique_ptr<TritonModelInstance>> passive_instances_;
+  // The model instances for this model stored by the
+  // instance groups defined in the model configuration.
+  // Passive instance groups are those instances which are
+  // loaded but not added to the scheduler.
+  std::map<std::string, TritonInstanceGroup> instance_group_map_;
+  std::map<std::string, TritonInstanceGroup> passive_instance_group_map_;
```

Comment:
Do the `map`s need to be ordered?

Reply:
See [this comment](https://github.com/triton-inference-server/core/pull/153#discussion_r1063758281)